crypto-ffi/src/generic/mod.rs

```diff
@@ -117,6 +117,16 @@ pub enum MlsError {
     /// with an External Commit instead
     #[error("Although this Welcome seems valid, the local KeyPackage it references has already been deleted locally. Join this group with an external commit")]
     OrphanWelcome,
+    /// This error is returned based on the abort reason returned in `MlsTransportResponse`. It is constructed when the reason is equal to `mls-client-mismatch`.
+    /// You should catch this error and restart the operation.
+    #[error("The message was rejected by the delivery service because there was a mismatch in mls clients. Restart the operation.")]
+    RestartOperation,
```

Comment:
```suggestion
    RetryOperation,
```

I'm not sure we should make decision for the client. Maybe it's better to keep as `OperationAborted` and let clients decide if they want to retry the operation or not.

Reply:
I was unsure about this, I'll just drop the corresponding commit and the extra error variant.